src/dom.js

```diff
@@ -418,6 +418,38 @@ export function hasNextNodeInDocumentOrder(element) {
 
 
 /**
+ * Finds all ancestor elements that satisfies the callback.
+ * @param {!Element} child
+ * @param {function(!Element):boolean} callback
+ * @return {!Array<!Element>}
+ */
+export function ancestorElements(child, callback) {
+  const ancestors = [];
+  for (let ancestor = child.parentElement; ancestor;
+       ancestor = ancestor.parentElement) {
+    if (callback(ancestor)) {
+      ancestors.push(ancestor);
+    }
+  }
+  return ancestors;
+}
+
+
+/**
+ * Finds all ancestor elements that has the specified tag name.
+ * @param {!Element} child
+ * @param {string} attr
+ * @return {!Array.<!Element>}
+ */
+export function ancestorElementsByTag(child, tagName) {
+  const upperTagName = tagName.toUpperCase();
```

Comment:
stay consistent with previous code where it just re-assign to param name (`tagName` in this case)